src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -358,16 +445,119 @@ private static IPHostEntry HostResolutionEndHelper(IAsyncResult asyncResult)
             return (IPHostEntry)castedResult.Result;
         }
 
-        private static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCallback requestCallback, object stateObject)
+        [Obsolete("BeginGetHostByName is obsoleted for this type, please use BeginGetHostEntry instead. http://go.microsoft.com/fwlink/?linkid=14202")]
+        public static IAsyncResult BeginGetHostByName(string hostName, AsyncCallback requestCallback, object stateObject)
+        {
+            if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostByName", hostName);
```

Comment:
In lots of cases, this would result in changes to the log (different capitalization, or in some cases different names).  We should consider doing this separately, across all uses of NetEventSource, GlobalLog, etc.